RecoCTPPS/PixelLocal/src/RPixRoadFinder.cc

```diff
@@ -0,0 +1,128 @@
+
+#include "RecoCTPPS/PixelLocal/interface/RPixRoadFinder.h"
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+
+//needed for the geometry:
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "TMatrixD.h"
+
+#include <vector>
+#include <memory>
+#include <string>
+#include <iostream>
+
+//------------------------------------------------------------------------------------------------//
+
+RPixRoadFinder::RPixRoadFinder(edm::ParameterSet const& parameterSet) : 
+	RPixDetPatternFinder(parameterSet),
+	param_(parameterSet){
+
+	verbosity_ = param_.getUntrackedParameter<int> ("RPixVerbosity");
+  roadRadius_ = param_.getParameter<double>("RPixRoadRadius");
+  minRoadSize_ = param_.getParameter<int>("RPixMinRoadSize");
+  maxRoadSize_ = param_.getParameter<int>("RPixMaxRoadSize");
+
+}
+
+//------------------------------------------------------------------------------------------------//
+
+RPixRoadFinder::~RPixRoadFinder(){
+}
+
+//------------------------------------------------------------------------------------------------//
+
+void RPixRoadFinder::findPattern(){
+
+  Road temp_all_hits;
+  temp_all_hits.clear();
+
+// convert local hit sto global and push them to a vector
+  for(const auto & ds_rh2 : hitVector_){
+    uint32_t myid = ds_rh2.id;
+    for (const auto & _rh : ds_rh2.data){
+      PointInPlane thePointAndRecHit;
+      thePointAndRecHit.recHit=_rh; 
+      CLHEP::Hep3Vector localV(_rh.getPoint().x(),_rh.getPoint().y(),_rh.getPoint().z() );
+      CLHEP::Hep3Vector globalV = geometry_.localToGlobal(ds_rh2.id,localV);
+      thePointAndRecHit.globalPoint=globalV;
+      TMatrixD localError(3,3);
+      localError[0][0] = _rh.getError().xx();
+      localError[0][1] = _rh.getError().xy();
+      localError[0][2] =                  0.;
+      localError[1][0] = _rh.getError().xy();
+      localError[1][1] = _rh.getError().yy();
+      localError[1][2] =                  0.;
+      localError[2][0] =                  0.;
+      localError[2][1] =                  0.;
+      localError[2][2] =                  0.;
+      if(verbosity_>2) std::cout<<"Hits = "<<ds_rh2.data.size()<<std::endl;
+      TMatrixD theRotationTMatrix(planeRotationMatrixMap_[CTPPSPixelDetId(ds_rh2.id)]);
```

Comment:
DataFormats/GeometryCommonDetAlgo/interface/ErrorFrameTransformer.h
may be of better use here; this would also allow using more common code.

Reply:
Dear Slava,
after discussing with Fabrizio, we understood that the method you are suggesting needs that DetGeom is included in the Geometry, but it is not for the one we are currently using.
The inclusion of these methods would require several days of work and we are wondering to eventually include them into a new PR if you think it is needed.
What is you opinion on that?